### DISCUSSION TITLE
React Native Payment Flow

### DISCUSSION THREAD

**Discussion by rossparachute** [dojo-engineering/feedback#15] [Docs]
@martinhay1 @RomanKondratyev

Hey folks,

We've got the Dojo React Native package hooked up.

Are we right in thinking that the payment intent id passed has to be for an intent we've created beforehand?
https://github.com/dojo-engineering/react-native-pay-sdk#usage

So, in a similar way to us initializing the checkout process for web by poking an endpoint we handle to create the intent that gets back a secret?
https://github.com/dojo-engineering/dojo-samples/blob/main/card-component/client/html/static/script.js#L17-L59
https://github.com/dojo-engineering/dojo-samples/blob/main/card-component/server/cs/server.cs

Then we pass it to the startPaymentFlow() args and the SDK and UI SDK handles the rest and spits out a response code?

Thanks again,
Ross

**Comment by deniss-kaibagarovs** [comment]
Hey @rossparachute, you're correct! Once you generated a payment-intent-id, you can pass it to the WEB or SDK. It would work regardless of the platform.
